bite-size-notes/week-4-e-stochastic bellman equation.md

```diff
@@ -0,0 +1,167 @@
+
+Overall task will be to determine what is the “optimal” control for our system. For this we need to specify performance for each controls. Let the state represent as a node and control sequences correspond to paths originating at the initial state and terminating at one of the goal state correspond to the Terminal stage $N$. So if we view the cost of path as its length, we see that a determinisitic finite state finite horizon problem is equivalent to finding a minimun-length path from the initial state of the environment to the terminal state.
+
+However, this is the deterministic optimazation problem. 
+
+Consider the finite-horizon discrete-time system which forms:
+
+$$
+x_{k+1} = f_k(x_k, u_k, \omega_k), \quad k=0,1,...,N-1
+$$
+
+where as $x_k$ is an element of some state space $S_k$, the control $u_k$ is an element of some control space. The cost per stage is denoted $g_k(x_k,u_k,\omega_k)$ and also depends on the random disturbance $\omega_k$. Then we can intepret transition graph as a Markov Decision Process. The control $u_k$ is constrained to take values in a given subset $U(x_k)$, which depends on the current state $x_k$.
+
+The stochastic finite horizon optimal control problem differs from the deterministic version primarily in the nature of the discrete-time dynamic system that governs the evolution of the state $x_k$. This evolution is also be called as the the state transition probability $p(x_k | x_{k−1} , u_k)$ where probability distribution assigns a probability or density value to each possible state with regards to the true state. Also, stochastic system includes a random "disturbance" $\omega_k$, which is characterized by a probability distribution $p(\omega_k|x_k, u_k)$ that may depend explicitly on $x_k$ and $u_k$, but not on values of prior disturbances $\omega_{k-1}, \omega_{k-2}, ...., \omega_0$. This is so called the __markov assumption__. In reality, this disturbance also contains random and contains uncertainties. Generally, disturbance are mostly originate from the observation, such that we can call as the measurement probability $p(z_k|x_{k}, u_{k-1}) = p(z_k|x_{k})$  where intuition can be given as the "noisy projection of the state". The state transition probability and the measurement probability together describe the dynamical stochastic system.
+
+
+
+
+## Major Difference between deterministic and stochastic optimal control
+An important difference is that we optimize not over control sequences ${u_0,u_1,...u_{N-1}}$, but rather over policies that consist of a sequence of functions, since the the stochastic dynamical system samples the estimated state information, control $u_k$ is chosen with knowledge of history of measurements(policies) related to the state.

```

Comment:
I like the conclusion that stochastic opt is optimize the policy rather than control.

Reply:
Thanks!